From: Olivier Fourdan <ofourdan@redhat.com>
Date: Tue, 14 Jun 2022 16:31:43 +0200
Subject: backends: Move MetaKeyboardA11yFlags to a public header

The MetaKeyboardA11yFlags are used by gnome-shell to show a dialog
whenever a keyboard accessibility feature is switched using the
keyboard.

Unfortunately, commit c3acaeb25 renamed the Clutter flag to Meta and
moved them to a private header. As a result, gnome-shell do not show any
dialog anymore when a keyboard accessibility feature is activated.

Move the MetaKeyboardA11yFlags definition to a public header so that
gnome-shell can use it.

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/2306
Fixes: c3acaeb25 - backends: Move keyboard a11y into backends
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2463>
(cherry picked from commit 1c16e1cb9c6b0c62dad4c318117324af6485fada)

Origin: upstream gnome-42 branch after 42.9
---
 src/backends/meta-input-settings-private.h | 19 +--------------
 src/meta/meta-enums.h                      | 38 ++++++++++++++++++++++++++++++
 2 files changed, 39 insertions(+), 18 deletions(-)

diff --git a/src/backends/meta-input-settings-private.h b/src/backends/meta-input-settings-private.h
index 42ee0e0..5ef54ee 100644
--- a/src/backends/meta-input-settings-private.h
+++ b/src/backends/meta-input-settings-private.h
@@ -31,29 +31,12 @@
 #include "backends/meta-backend-types.h"
 #include "clutter/clutter.h"
 #include "meta/display.h"
+#include "meta/meta-enums.h"
 
 #define META_TYPE_INPUT_SETTINGS (meta_input_settings_get_type ())
 G_DECLARE_DERIVABLE_TYPE (MetaInputSettings, meta_input_settings,
                           META, INPUT_SETTINGS, GObject)
 
-typedef enum
-{
-  META_A11Y_KEYBOARD_ENABLED = 1 << 0,
-  META_A11Y_TIMEOUT_ENABLED = 1 << 1,
-  META_A11Y_MOUSE_KEYS_ENABLED = 1 << 2,
-  META_A11Y_SLOW_KEYS_ENABLED = 1 << 3,
-  META_A11Y_SLOW_KEYS_BEEP_PRESS = 1 << 4,
-  META_A11Y_SLOW_KEYS_BEEP_ACCEPT = 1 << 5,
-  META_A11Y_SLOW_KEYS_BEEP_REJECT = 1 << 6,
-  META_A11Y_BOUNCE_KEYS_ENABLED = 1 << 7,
-  META_A11Y_BOUNCE_KEYS_BEEP_REJECT = 1 << 8,
-  META_A11Y_TOGGLE_KEYS_ENABLED = 1 << 9,
-  META_A11Y_STICKY_KEYS_ENABLED = 1 << 10,
-  META_A11Y_STICKY_KEYS_TWO_KEY_OFF = 1 << 11,
-  META_A11Y_STICKY_KEYS_BEEP = 1 << 12,
-  META_A11Y_FEATURE_STATE_CHANGE_BEEP = 1 << 13,
-} MetaKeyboardA11yFlags;
-
 /**
  * MetaKbdA11ySettings:
  *
diff --git a/src/meta/meta-enums.h b/src/meta/meta-enums.h
index 373d144..1cf9a91 100644
--- a/src/meta/meta-enums.h
+++ b/src/meta/meta-enums.h
@@ -27,4 +27,42 @@ typedef enum _MetaCompositorType
   META_COMPOSITOR_TYPE_X11,
 } MetaCompositorType;
 
+/**
+ * MetaKeyboardA11yFlags:
+ * @META_A11Y_KEYBOARD_ENABLED:
+ * @META_A11Y_TIMEOUT_ENABLED:
+ * @META_A11Y_MOUSE_KEYS_ENABLED:
+ * @META_A11Y_SLOW_KEYS_ENABLED:
+ * @META_A11Y_SLOW_KEYS_BEEP_PRESS:
+ * @META_A11Y_SLOW_KEYS_BEEP_ACCEPT:
+ * @META_A11Y_SLOW_KEYS_BEEP_REJECT:
+ * @META_A11Y_BOUNCE_KEYS_ENABLED:
+ * @META_A11Y_BOUNCE_KEYS_BEEP_REJECT:
+ * @META_A11Y_TOGGLE_KEYS_ENABLED:
+ * @META_A11Y_STICKY_KEYS_ENABLED:
+ * @META_A11Y_STICKY_KEYS_TWO_KEY_OFF:
+ * @META_A11Y_STICKY_KEYS_BEEP:
+ * @META_A11Y_FEATURE_STATE_CHANGE_BEEP:
+ *
+ * Keyboard accessibility features.
+ *
+ */
+typedef enum
+{
+  META_A11Y_KEYBOARD_ENABLED = 1 << 0,
+  META_A11Y_TIMEOUT_ENABLED = 1 << 1,
+  META_A11Y_MOUSE_KEYS_ENABLED = 1 << 2,
+  META_A11Y_SLOW_KEYS_ENABLED = 1 << 3,
+  META_A11Y_SLOW_KEYS_BEEP_PRESS = 1 << 4,
+  META_A11Y_SLOW_KEYS_BEEP_ACCEPT = 1 << 5,
+  META_A11Y_SLOW_KEYS_BEEP_REJECT = 1 << 6,
+  META_A11Y_BOUNCE_KEYS_ENABLED = 1 << 7,
+  META_A11Y_BOUNCE_KEYS_BEEP_REJECT = 1 << 8,
+  META_A11Y_TOGGLE_KEYS_ENABLED = 1 << 9,
+  META_A11Y_STICKY_KEYS_ENABLED = 1 << 10,
+  META_A11Y_STICKY_KEYS_TWO_KEY_OFF = 1 << 11,
+  META_A11Y_STICKY_KEYS_BEEP = 1 << 12,
+  META_A11Y_FEATURE_STATE_CHANGE_BEEP = 1 << 13,
+} MetaKeyboardA11yFlags;
+
 #endif /* META_ENUMS_H */
